src/index.ts

```diff
@@ -45,10 +45,17 @@ const indexer = (
   const transform = (documents: SanityDocumentStub[]) => {
     const records: AlgoliaRecord[] = documents.map(
       (document: SanityDocumentStub) => {
-        return Object.assign(standardValues(document), serializer(document))
+        const serializedDocs = serializer(document)
+        if (Array.isArray(serializedDocs)) {
+          return serializedDocs.map((chunk) =>
+            Object.assign(standardValues(chunk), chunk)
+          )
+        } else {
+          return Object.assign(standardValues(document), serializedDocs)
+        }
```

Comment:
if the object passed back from `serializer` is an array, then we know we need to map the objects inside and assign the standard values.